### PR TITLE
10 create POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by w1855014**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -24,6 +24,10 @@ app.use((err, req, res, next) =>
     {
         res.status(400).send({msg: "Bad request."});
     }
+    if (err.code === "23503")
+    {
+        res.status(404).send({msg: "Not found."})
+    }
     next(err);
 })
```
Comment (elijaharley) [inline app.js:33]: Nice custom error handling here for PSQL errors

File: models/articles.js [hunk 2]
```diff
@@ -78,7 +78,23 @@ exports.incrementArticleVotesById = (id, inc_votes) =>
...
+        console.log(rows);
+        if (!rows.length)
+        {
+            const err = new Error(`Not found.`);
+            err.status = 404;
+            return Promise.reject(err);
+        }
+        return rows[0];
+    });
 }
```
Comment (elijaharley) [inline models/articles.js:100]: Don't forget to remove console.logs once you've finished using them

File: routes/articles.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 
-const {getAllArticles, getArticleById, patchArticleVotesById} = require('../controllers/articles');
+const {getAllArticles, getArticleById, patchArticleVotesById, postCommentByArticleId} = require('../controllers/articles');
 
 router.get('/', getAllArticles);
 router.get('/:article_id', getArticleById);
 router.patch('/:article_id', patchArticleVotesById);
+router.post('/:article_id/comments', postCommentByArticleId);
 
 module.exports = router;
```
Comment (elijaharley) [inline routes/articles.js:11]: Great to see that you've managed to implement routing here

File: __tests__/app.test.js [hunk 4]
```diff
@@ -136,6 +136,55 @@ describe('articles', () =>
...
+            .send({username: "icellusedkars", body: "Amazing."})
+            .expect(404)
+            .then(({body}) =>
+            {
+                const {msg} = body;
+                expect(msg).toBe("Not found.")
+            })
+        });
     })
 });
```
Comment (elijaharley) [inline __tests__/app.test.js:190]: Make sure your test descriptions are describing each specific endpoint eg '404: returns not found when passed a non existent article'

I would also add a test here for an empty body.

**Review by elijaharley — APPROVED** [review]
Really nice work on this. See comments about test descriptions and then you're good to merge.
